pkg/component/etcd/service/service.go

```diff
@@ -98,6 +98,18 @@ func getOwnerReferences(val Values) []metav1.OwnerReference {
 	}
 }
 
+func getSelectors(val Values) map[string]string {
+	selectors := map[string]string{
+		"instance": val.EtcdName,
+	}
+
+	for k, v := range val.Selectors {
+		selectors[k] = v
+	}
```

Comment:
I wonder if it's really beneficial to have additional selectors defined by the user while we already have a well known and unique selector with `"instance": val.EtcdName`.

Reply:
I agree it is not required.

Reply:
In my opinion, we can keep it. If the users don't want to put any other unique selector then they may choose to not mention any in etcd.spec.selector, right?

Reply:
From a gardener perspective it is not required. From an open source perspective if you wish to open it up (and not assume any default labels and selectors) then you need to also handle the following:

1. A webook to validate if the labels and selectors are proper, else the pods will never be discovered by the respective services.
2. Ensure that if lable-selector has been changed in a manner that requires a recreation of STS then it needs to be handled in component/statefulset

Reply:
> If the users don't want to put any other unique selector then they may choose to not mention any in etcd.spec.selector, right?

Can you elaborate on this? I'd like to understand the use case of defining additional selectors via the `etcd` resource.

Reply:
Currently, it is not required. I was talking mainly from open source point of view. I was thinking, let the code be in place, if somebody require to add something in future, they can do it at their choice.

Reply:
As written in chat:

There are however many downsides:
- Re-creation of STS is necessary
- Incompatible change
- Misusage of API field (earlier it was used to adopt STS … now it’s used for service selector)

In my eyes, we should only follow the approach in this PR if there is a valid and beneficial use-case for the user. So far I don't see a justification to accept the downsides listed above.

Reply:
Made the changes

Reply:
I don't see the changes being made. The service selector still considers `etcd.Spec.Selector.MatchLabels`.